app/scripts/utils/track-utils.js

```diff
@@ -0,0 +1,404 @@
+ * the minX and maxX values for the region
+ *
+ * @param resolution: The number of base pairs per bin
+ * @param scale: The scale to use to calculate the currently visible tiles
+ * @param minX: The minimum x position of the tileset
+ * @param maxX: The maximum x position of the tileset
+ */
+const calculateTilesFromResolution = (resolution, scale, minX, maxX, pixelsPerTile) => {
+  const epsilon = 0.0000001;
+  const PIXELS_PER_TILE = pixelsPerTile || 256;
+  const tileWidth = resolution * PIXELS_PER_TILE;
+  const MAX_TILES = 20;
+  // console.log('PIXELS_PER_TILE:', PIXELS_PER_TILE);
+
+  if (!maxX) {
+    maxX = Number.MAX_VALUE; // eslint-disable-line no-param-reassign
+  }
+
+  const lowerBound = Math.max(0, Math.floor((scale.domain()[0] - minX) / tileWidth));
+  const upperBound = Math.ceil(Math.min(
+    maxX,
+    ((scale.domain()[1] - minX) - epsilon)
+  ) / tileWidth);
+  let tileRange = range(
+    lowerBound,
+    upperBound,
+  );
+
+  if (tileRange.length > MAX_TILES) {
+    // too many tiles visible in this range
+    console.warn(`Too many visible tiles: ${tileRange.length} truncating to ${MAX_TILES}`);
+    tileRange = tileRange.slice(0, MAX_TILES);
+  }
+  // console.log('tileRange:', tileRange);
+
+  return tileRange;
+};
+
+/**
+ * Calculate which tiles should be visible given a track's
+ * scale.
+ *
+ * @param  {Object} tilesetInfo The track's tileset info, containing either the `resolutions`
+ *                              list or min_pos and max_pos arrays
+ * @param  {d3.scale} scale     The track's scale.
+ * @return {Array}             A list of visible tiles (e.g. [[1,0],[1,1]])
+ */
+const calculate1DVisibleTiles = (tilesetInfo, scale) => {
+  // if we don't know anything about this dataset, no point
+  // in trying to get tiles
+  if (!tilesetInfo) { return []; }
+
+  // calculate the zoom level given the scales and the data bounds
+  const zoomLevel = calculate1DZoomLevel(tilesetInfo, scale, tilesetInfo.max_zoom);
+
+  if (tilesetInfo.resolutions) {
+    const sortedResolutions = tilesetInfo.resolutions
+      .map(x => +x)
+      .sort((a, b) => b - a);
+
+    const xTiles = calculateTilesFromResolution(
+      sortedResolutions[zoomLevel],
+      scale,
+      tilesetInfo.min_pos[0], tilesetInfo.max_pos[0]
+    );
+
+    const tiles = xTiles.map(x => [zoomLevel, x]);
+
+    return tiles;
+  }
+
+  // x doesn't necessary mean 'x' axis, it just refers to the relevant axis
+  // (x if horizontal, y if vertical)
+  const xTiles = calculateTiles(
+    zoomLevel,
+    scale,
+    tilesetInfo.min_pos[0],
+    tilesetInfo.max_pos[0],
+    tilesetInfo.max_zoom,
+    tilesetInfo.max_width
+  );
+
+  const tiles = xTiles.map(x => [zoomLevel, x]);
+  return tiles;
+};
+
+/**
+ * Draw an axis on a track. Where on the track the axis will be drawn
+ * is taken from the track's options.
+ *
+ * @param  {PixiTrack} track   The track to decorate with an axis.
```

Comment:
```suggestion
 * @param  {object} track   The track instance to decorate with an axis.
```

I assume it's the track instance is that correct? Or should I (as a developer) pass in the track constructor function?

Reply:
Track instance is correct :-)